src/Domain/EmbeddingMaps/AffineMap.hpp

```diff
@@ -0,0 +1,97 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines the class AffineMap.
+
+#pragma once
+
+#include <memory>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/EmbeddingMaps/EmbeddingMap.hpp"
+#include "Parallel/CharmPupable.hpp"
+
+namespace EmbeddingMaps {
+
+/// Linear map from \f$\xi \in [A, B]\rightarrow x \in [a, b]\f$.
+/// The formula for the mapping is...
+/*! \f[
+  x = \frac{b}{B-A} (\xi-A) +\frac{a}{B-A}(B-\xi)
+ \f]
+ \f[
+  \xi =\frac{B}{b-a} (x-a) +\frac{A}{b-a}(b-x)
+ \f]
+*/
+class AffineMap : public EmbeddingMap<1, 1> {
+ public:
+  AffineMap(const double A, const double B, const double a, const double b)
+      : A_(A), B_(B), a_(a), b_(b), L_(B - A), l_(b - a) {}
+
+  AffineMap() = default;
+  ~AffineMap() override = default;
+  AffineMap(const AffineMap&) = delete;
+  AffineMap(AffineMap&&) noexcept = default;  // NOLINT
+  AffineMap& operator=(const AffineMap&) = delete;
+  AffineMap& operator=(AffineMap&&) = default;
+
+  std::unique_ptr<EmbeddingMap<1, 1>> get_clone() const override {
+    return std::make_unique<AffineMap>(A_, B_, a_, b_);
+  }
+
+  Point<1, Frame::Grid> operator()(
+      const Point<1, Frame::Logical>& xi) const override {
+    return Point<1, Frame::Grid>((l_ * xi[0] + a_ * B_ - b_ * A_) / L_);
+  }
+
+  Point<1, Frame::Logical> inverse(
+      const Point<1, Frame::Grid>& x) const override {
+    return Point<1, Frame::Logical>((L_ * x[0] - a_ * B_ + b_ * A_) / l_);
+  }
+
+  double jacobian(const Point<1, Frame::Logical>& /* xi */, size_t ud,
+                  size_t ld) const override {
+    ASSERT(0 == ld, "ld = " << ld);
+    ASSERT(0 == ud, "ud = " << ud);
+    return l_ / L_;
+  }
+
+  double inv_jacobian(const Point<1, Frame::Logical>& /*xi*/, size_t ud,
+                      size_t ld) const override {
+    ASSERT(0 == ld, "ld = " << ld);
+    ASSERT(0 == ud, "ud = " << ud);
+    return L_ / l_;
+  }
+
+  WRAPPED_PUPable_decl_base_template(SINGLE_ARG(EmbeddingMap<1, 1>),  // NOLINT
+                                     AffineMap);
+
+  explicit AffineMap(CkMigrateMessage* /* m */)
+      : A_(-1), B_(1), a_(-1), b_(1), L_(B_ - A_), l_(b_ - a_) {}
```

Comment:
I just read up on it. In all cases we have so far it should behave either exactly as the default constructor or set everything to "evil" values that will cause an abort. Basically, what happens when a chare array element is migrated or a PUP::able object is migrated the migration constructor is called, followed by the `pup` method. The `pup` method is used to transfer data while the migration constructor could be used to recompute constant data that does not need to be serialized and where recomputing is faster than serialization.